src/core/LocalBox.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef ESPRESSO_LOCALBOX_HPP
+#define ESPRESSO_LOCALBOX_HPP
+
+#include <utils/Vector.hpp>
+
+template <class T> class LocalBox {
+public:
```

Comment:
wouldn't it be cleaner to make the members private and have a constructor such that an instance of ```LocalBox``` cannot be in an inconsistent state